sdk/provisioning/Azure.Provisioning.CloudMachine/src/AzureSdkExtensions/OpenAIFeature.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.Diagnostics.Contracts;
+using Azure.AI.OpenAI;
+using Azure.CloudMachine;
+using Azure.Core;
+using Azure.Provisioning.Authorization;
+using Azure.Provisioning.CognitiveServices;
+using OpenAI.Chat;
+
+namespace Azure.Provisioning.CloudMachine.OpenAI;
+
+public class OpenAIFeature : CloudMachineFeature
+{
+    public string Model { get; }
+    public string ModelVersion { get; }
+
+    public OpenAIFeature(string model, string modelVersion) {  Model = model; ModelVersion = modelVersion; }
+
+    public override void AddTo(CloudMachineInfrastructure infrastructure)
+    {
+        CognitiveServicesAccount cognitiveServices = new("openai", "2023-05-01")
```

Comment:
Nit - you can either drop the versions to float on latest (which is already `2024-10-01`) or use `CognitiveServicesAccount.ResourceVersions.V2023_05_01` if you don't want to deal with the strings to pin.